usaspending_api/references/v2/views/recipients.py

```diff
@@ -0,0 +1,110 @@
+from collections import OrderedDict
+
+from elasticsearch_dsl import Q as ES_Q
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.elasticsearch.search_wrappers import RecipientSearch
+from usaspending_api.common.validator.tinyshield import validate_post_request
+from usaspending_api.search.v2.es_sanitization import es_sanitize
+
+models = [
+    {"name": "search_text", "key": "search_text", "type": "text", "text_type": "search", "optional": False},
+    {"name": "limit", "key": "limit", "type": "integer", "max": 500, "optional": True, "default": 10},
+    {
+        "name": "recipient_levels",
+        "key": "recipient_levels",
+        "type": "array",
+        "array_type": "text",
+        "text_type": "search",
+        "items": {"type": "string"},
+        "optional": True,
+    },
+]
+
+
+@validate_post_request(models)
+class RecipientAutocompleteViewSet(APIView):
+    """
+    This endpoint is used for the Recipient autocomplete filter which returns a list of recipients matching the
+    specified search text.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/autocomplete/recipient.md"
+
+    @cache_response()
+    def post(self, request, format=None):
+        search_text, recipient_levels = prepare_search_terms(request.data)
+        limit = request.data["limit"]
+        query = create_es_search(search_text, recipient_levels, limit)
+        results = query_elasticsearch(query)
+        response = OrderedDict([("count", len(results)), ("results", results)])
+        return Response(response)
+
+
+def prepare_search_terms(request_data):
```

Comment:
> This method probably should be a behavior of the class RecipientAutocompleteViewSet and not just a function belonging to this module. Additionally, you can probably make this method private via a leading underscore.

Reply:
What do we mean by "a behavior of the class RecipientAutocompleteViewSet and not just a function belonging to this module"?

Reply:
When we talk about object oriented programming, "behaviors" mean methods of a class. To be exact, a behavior is something an object does. So a method is a behavior of a class.

Reply:
Just make them methods of the class not functions.